text/0000-roadmap-2019.md

```diff
@@ -0,0 +1,328 @@
+>  
+> - Raph Levien, ["My thoughts on Rust 2019"](https://raphlinus.github.io/rust/2018/12/16/rust-2019.html)
+
+> If you’re familiar with the history of Rust you might be confused with a
+> call to stabilization. After all, Rust 2015 (a.k.a 1.0) was all about
+> stabilization and the team has actually done a pretty good job with achieving
+> this goal. So what gives?
+> 
+> While Rust 2015 defined stabilization around language backward
+> compatibility, it’s time for the language, the tooling, the ecosystem and the
+> governance to stabilize. Each of these characterize stability in a different
+> way.
+> 
+> - Ryan Levick, ["Rust 2019: Stabilization"](https://blog.ryanlevick.com/posts/rust-2019/)
+
+> The Rust project has been growing like a startup for the last several
+> years. This has some good aspect - “hockeystick curves” - but it has some
+> very bad aspects as well. If this project is going to have a chance of
+> sustaining itself in the long term, we need to get real about dealing with
+> the organizational debt we have accumulated. I think we have serious problems
+> at every level of our organization that need to be addressed, and I’m going
+> to enumerate them from my personal perspective.
+> 
+> - withoutboats, ["Organizational Debt"](https://boats.gitlab.io/blog/post/rust-2019/)
+
+We don't have space to quote every single post; if you'd like to read them
+all, you can over at [Read Rust](https://readrust.net/rust-2019/). And of
+course, these themes are not *universally* shared among Rust developers, but
+there is a common theme in many of them: slow down, pay off debt, finish what
+we've started.
+
+# By team
+[By team]: #by-team
+
+With this general theme in mind, here are the high-level plans from each of
+the teams for the upcoming year, in alphabetical order. These summaries are
+based on discussions had at the All Hands; each team may also post a more
+detailed individual roadmap. The summaries are meant to show plans in broad
+strokes and how they fit together for the project.
+
+## Cargo
+
+The [cargo team 2019 roadmap](https://gist.github.com/nrc/6f0fb3b66d3b043aace30217128f3af9) identifies a few major themes:
+
+- **Better support for cross compilation:**
+    - Try to incorporate some of the innovations from tools like [Xargo](https://github.com/japaric/xargo), [cross](https://github.com/rust-embedded/cross), [Rustup](https://github.com/rust-lang/rustup.rs), and [WASM-Pack](https://github.com/rustwasm/wasm-pack) into cargo itself.
+- **Focus on the cargo ecosystem by supporting plugins (custom commands):**
+    - Custom commands enable faster experimentation, bringing new and flexible workflows to users sooner rather than later.
+    - Make plugins more discoverable.
+    - Provide library crates for parts of cargo to make it easier to create plugins.
+- **Compile times:**
+    - Investigate ways that cargo can help to reduce compilation time, including potentially distributing binaries for builds from crates.io.
```

Comment:
That would mean supporting binaries for a lot of targets (which is ok IMO).  There can also be a target "MIR" where people could build crates from the MIR code, saving some time in parsing/type-checking/borrow-checking*.  This could be done with a fallback mechanism: is there a binary matching my target? if not use MIR, if that doesn't exist as well build from source.

Reply:
Of course there is the problem of conditional compilation (`#[cgf(...)]`)...

Reply:
It also makes me a bit wary security-wise, but I guess that can be discussed at the time...